plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableQueryTests.php

```diff
@@ -235,4 +236,43 @@ public function test_query_filters() {
 		remove_all_filters( 'woocommerce_orders_table_query_clauses' );
 	}
 
+	/**
+	 * @testdox The pre-query escape hook allows replacing the order query.
+	 */
+	public function test_pre_query_escape_hook() {
+		$order1 = new \WC_Order();
+		$order1->set_date_created( time() - HOUR_IN_SECONDS );
+		$order1->save();
+
+		$order2 = new \WC_Order();
+		$order2->save();
+
+		$query = new OrdersTableQuery( array() );
+		$this->assertCount( 2, $query->orders );
+		$this->assertEquals( 2, $query->found_orders );
+		$this->assertEquals( 0, $query->max_num_pages );
+
+		$callback = function( $query ) use ( $order1 ) {
+			// Only return one of the orders to show that we are replacing the query result.
+			$order_ids = array( $order1->get_id() );
+			// These are made up to show that we are actually replacing the values. 
+			$found_orders = 17;
+			$max_num_pages = 23;
+			return [ $order_ids, $found_orders, $max_num_pages ];
```

Comment:
I thought we use the `array( ` syntax, but phpcs is not failing, so maybe that's updated.

Reply:
I don't think PHPCS is running correctly (here in GitHub). Let's be sure to lint locally until things are fixed.

Reply:
I ran the linter locally and it didn't mind this syntax. I wonder if increasing the PHP version could be the reason? 

https://github.com/woocommerce/woocommerce/pull/39820

Reply:
Managed to run `pnpm lint:php:changes:branch` and fixed all found issues -- my local PHP was set to 8.2, which I remembered doesn't work with `phpcs` at this time. ✅